classes/cargo.bbclass

```diff
@@ -116,12 +120,23 @@ cargo_do_compile() {
     export TARGET_CXX="${WRAPPER_DIR}/cxx-wrapper.sh"
     export CC="${WRAPPER_DIR}/cc-native-wrapper.sh"
     export CXX="${WRAPPER_DIR}/cxx-native-wrapper.sh"
-    export TARGET_LD="${WRAPPER_DIR}/ld-wrapper.sh"
-    export LD="${WRAPPER_DIR}/ld-native-wrapper.sh"
     export PKG_CONFIG_ALLOW_CROSS="1"
     export LDFLAGS=""
     export RUSTFLAGS="${RUSTFLAGS}"
     export SSH_AUTH_SOCK="${SSH_AUTH_SOCK}"
+    # When RUST_BUILD == RUST_TARGET, we need to use an unstable Rust feature
+    # to specify different build options for the target and the Host.
+    # The Host configuration is set in do_configure() above.
+    if [ "${RUST_BUILD}" = "${RUST_TARGET}" ]; then
```

Comment:
Probably best to remove this check and just unconditionally apply these env variables like we do in buildroot.

Reply:
Will do.